### DISCUSSION TITLE
openupgrade_framework showing up in unrelated databases

### DISCUSSION THREAD

**Discussion by Robrechtc** [OCA/OpenUpgrade#3640] [General]
I have migrated a database from v14 to v15 a while back. 
This migration ran in a separate environment with its own interpreter, but my Odoo sources are global (I have one folder for each version that has the source for community and enterprise, instead of one repository for Odoo per project). 

I believe that OpenUpgrade sets a server_wide_modules flag somewhere in a default Odoo configuration or environment variable. 
It is my opinion that the openupgrade_framework should be removed from the server_wide_modules whenever a migration is complete, as this can interfere with other projects on a machine (in my case, bricking a v16 install and leading to a very annoying search for what caused it and why the openupgrade_framework was interfering).

**Comment by legalsylvain** [comment]
Hi. 
I don't understand your point. the server wide modules is set by the command line (odoo-bin --load) or by the odoo.cfg file. In both situation, it's up to you. don't you think ?

**Reply by Robrechtc** [reply]
If I use a simple config with only db_name and addons_path (so without the server_wide_modules), Odoo will try to load the openupgrade_framework: 

![image](https://user-images.githubusercontent.com/2767030/206650443-7165cb5f-f970-47b6-a036-30a830daf066.png)

I'm using this to start my local instance: 

![image](https://user-images.githubusercontent.com/2767030/206650522-1ff60aff-d653-48db-a709-7bd1c38a32ee.png)

If I add the server_wide_modules to my config, it works properly, but I shouldn't have to - it is only mandatory now because if I don't there's a default setting that also tries to load the openupgrade_framework.

**Comment by legalsylvain** [comment]
> If I add the server_wide_modules to my config, it works properly, but I shouldn't have to 

Hi. This is how openupgrade works. you should write --load=openupgrade_framework / server_wide_modules=openupgrade_framework

regards.

**Comment by Robrechtc** [comment]
Sylvain,

I am aware - I mean that now, for whatever reason, with other projects and
even other Odoo versions on the same machine I have to specify
server_wide_modules=False to stop OpenUpgrade from loading (for those other
instances I do not want to use OpenUpgrade).

Op di 3 jan. 2023 16:36 schreef Sylvain LE GAL ***@***.***>:
